framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -88,35 +104,36 @@ PatternedMeshGenerator::generate()
                  type(),
                  " only works with inputs that are replicated.\n\n",
                  "Try running without distributed mesh.");
-    _meshes.push_back(std::move(mesh));
-  }
-
-  // Data structure that holds each row
-  _row_meshes.resize(_pattern.size());
-
-  // Getting the boundaries provided by the user
-  std::vector<BoundaryName> boundary_names = {getParam<BoundaryName>("left_boundary"),
-                                              getParam<BoundaryName>("right_boundary"),
-                                              getParam<BoundaryName>("top_boundary"),
-                                              getParam<BoundaryName>("bottom_boundary")};
+    _meshes[i] = dynamic_pointer_cast<ReplicatedMesh>(mesh->clone());
 
-  std::vector<boundary_id_type> ids =
-      MooseMeshUtils::getBoundaryIDs(*_meshes[0], boundary_names, true);
+    const auto ids = MooseMeshUtils::getBoundaryIDs(*_meshes[i], boundary_names, false);
+    mooseAssert(ids.size() == boundary_names.size(),
+                "Unexpected number of ids returned for MooseMeshUtils::getBoundaryIDs");
 
-  mooseAssert(ids.size() == boundary_names.size(),
-              "Unexpected number of ids returned for MooseMeshUtils::getBoundaryIDs");
+    // Keep track of used IDs so we can find IDs that are unused across all meshes
+    used_boundary_ids.insert(ids.begin(), ids.end());
 
-  boundary_id_type left = ids[0], right = ids[1], top = ids[2], bottom = ids[3];
+    // Check if all the boundaries have been initialized
+    for (const auto side : make_range(4))
+    {
+      if (ids[side] == Moose::INVALID_BOUNDARY_ID)
+        paramError("inputs",
+                   "The '",
+                   boundary_param_names[side],
+                   "' parameter with value '",
+                   boundary_names[side],
+                   "' does not exist in input mesh '",
+                   _input_names[i],
+                   "'");
```

Comment:
@loganharbour, let me know if you want any additional changes to this one. It's a bit tricky because an error results from a combination of TWO input parameters (i..e, when one of left_boundary, right_boundary, top_boundary, or bottom_boundary does not match up with a boundary name of any of the meshes in inputs).

Reply:
See comment above about generate_unknown parameter in getBoundaryIDs function.

Reply:
Talk to me about this in person, I'm a bit confused at what you're trying to say in the first comment. Also - we might want to check to make sure that all of the options in [left, right, top, bottom] are unique.